src/platform/mt8195/include/platform/lib/dma.h

```diff
@@ -0,0 +1,34 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ *
+ * Copyright(c) 2021 Mediatek
+ *
+ * Author: Bo Pan <bo.pan@mediatek.com>
+ */
+
+#ifdef __SOF_LIB_DMA_H__
+
+#ifndef __PLATFORM_LIB_DMA_H__
+#define __PLATFORM_LIB_DMA_H__
+
+#define PLATFORM_NUM_DMACS	2
+
+/* max number of supported DMA channels */
+#define PLATFORM_MAX_DMA_CHAN	32
+
+#define DMA_ID_AFE_MEMIF	0
+#define DMA_ID_HOST 1
+
+#define dma_chan_irq(dma, chan) dma_irq(dma)
+#define dma_chan_irq_name(dma, chan) dma_irq_name(dma)
+
+#define MEMIF_IRQ	32
+#define MEMIF_IRQ_NAME	"afe_memif_irq"
```

Comment:
The way IRQ names work in SOF is unintuitive; here you must give the name of the IRQ controller which has the given interrupt. The way this is resolved is it searches for the controller name here, then gets the interrupt (here you said 32) from that. Are you sure this is correct?

If there's no translation needed for the IRQ, the NAME must be set to NULL (or to the empty string, which for some reason is treated the same)